docs/scos/dev/feature-walkthroughs/202204.0/configurable-product-feature-walkthrough/configurable-product-flow-chart.md

```diff
@@ -0,0 +1,198 @@
+| `ProductConfigurationInstance.configuratorKey` | `dateTime` |   |
+| `ProductConfigurationInstance.isComplete` | `true` or `false` | Sensitive data. |
+| `backUrl` | `https://some.url` | Sensitive data. |
+| `sourceType` | `pdp`, `cart-page` |   |
+| `SKU` | `some_sku` |   |
+| `Quantity` | `2` |   |
+| `ItemGroupKey` | `some_key` |   |
+
+### Phase 2
+
+1. The customer clicks the configuration button,
+2. The form request is redirected to the gateway page with a given product configuration using the `GET` method.
+
+### Phase 3
+
+1. The configuration data is read from the quote for the given `ItemGroupKey` and `SKU`.
+2. A plugin that handles the request is selected based on the configurator type.
+3. The plugin expands the request with additional necessary data:
+  - store, locale, currency, customer, price mode.
+  - `CheckSum`—calculates the CRC32 polynomial of a request data as a string.
+  - `timestamp`—the timestamp when the request is created.
+  - `backUrl`—based on the `Referer` header.
+4. The plugin generates the URL that points to the configurator page together with all the necessary data.
+
+### Phase 4
+
+Redirects the customer to the configurator page using the GET request.
+
+### Phase 5
+
+1. We are on the configurator page.
+2. The customer does the configuration.
+3. Submits the configuration form.
+4. The configurator has to redirect to the gateway page with configuration data.
+
+| PARAMETER | VALUE | COMMENT |
+|---|---|---|
+| `ProductConfigurationInstance.price` | `2123123` | Sensitive data. |
+| `ProductConfigurationInstance.isComplete` | `1` | Sensitive data. |
+| `ProductConfigurationInstance.availableQuantity` | `2` |   |
+| `ProductConfigurationInstance.displayDescription` | Some text of JSON blob—for example, `["color"=>"red", "weight"=> 100]` |   |
+| `ProductConfigurationInstance.configuration` |  `"date"=>"23.07.2020", "time"=>"18:45" 4]`   | Sensitive data. |
+| `ProductConfigurationInstance.timestamp` | `10312313135234` | Sensitive data, a certain configuration should be valid only a certain amount of the time given. |
+| sourceType | pdp, cart-page, … |   |
+| SKU  | `some_sku` |   |
+| itemGroupKey | some_group_key |   |
+
+### Phase 6
+
+* After the successful configuration, the customer is redirected to the configurator gateway with a configuration response.
+* The gateway URL does not equal to the back URL, it’s a fixed known URL.
+
+### Phase 7
+
+1. The gateway page receives data.
+2. The data is checked through the execution of the validator plugins stack for received data.
+  - If validation is not successful, the request is redirected to the `backUrl` without saving the of the configuration with a warning message.
+  - If validation is successful, then updates the cart item configuration in the cart.
+4. All applicable plugins that can handle the configurator response are executed.
+  - A plugin that applies for the cart page source type resolves the back URL according to the response data.
```

Comment:
Is it a substep or we should make it step 5?
```suggestion
5. A plugin that applies for the cart page source type resolves the back URL according to the response data.
```

Reply:
It's part of 4, one of the plugins does that.